File: sol/solution/0001-0100/0019/Solution.scala
/**
 * Definition for singly-linked list.
 * class ListNode(_x: Int = 0, _next: ListNode = null) {
 *   var next: ListNode = _next
 *   var x: Int = _x
 * }
 */
object Solution {
    def removeNthFromEnd(head: ListNode, n: Int): ListNode = {
        var size = 0
        var iter = head
        while (iter != null) {
            iter = iter.next
            size += 1
        }
        var nth = size - n
        var i = 0
        iter = head
        while (i < nth - 1) {
            i += 1
            iter = iter.next
        }
        if (nth == 0) head.next
        else {
            iter.next = iter.next.next
            head
        }
    }
}

File: sol/solution/0001-0100/0019/Solution.rs
// Definition for singly-linked list.
// #[derive(PartialEq, Eq, Clone, Debug)]
// pub struct ListNode {
//   pub val: i32,
//   pub next: Option<Box<ListNode>>
// }
// 
// impl ListNode {
//   #[inline]
//   fn new(val: i32) -> Self {
//     ListNode {
//       next: None,
//       val
//     }
//   }
// }
use crate::list_node::ListNode;
impl Solution {
    pub fn remove_nth_from_end(head: Option<Box<ListNode>>, n: i32) -> Option<Box<ListNode>> {
        let mut dummy = Box::new(ListNode { val: 0, next: head });
        let mut len = 0;
        {
            let mut p = dummy.next.as_ref();
            while let Some(node) = p {
                len += 1;
                p = node.next.as_ref();
            }
        }
        let mut cur = &mut dummy;
        for _ in 0..(len - n) {
            cur = cur.next.as_mut().unwrap();
        }
        let next = cur.next.as_mut().and_then(|node| node.next.take());
        cur.next = next;
        dummy.next
    }
}
impl __serializer__::Serialize for ListNode {
    fn serialize(&self) -> String {
        let mut res = Vec::new();
        let mut node = Some(self);
        while let Some(n) = node {
            res.push(n.val.to_string());
            node = n.next.as_deref();
        }
        format!("[{}]", res.join(","))
    }
}